src/Streamly/Streams/StreamD.hs

```diff
@@ -1881,6 +1899,66 @@ isPrefixOf (Stream stepa ta) (Stream stepb tb) = go (ta, tb, Nothing)
             Skip sb' -> go (sa, sb', Just x)
             Stop     -> return False
 
+-- XXX Add tests and benchmarks
+-- XXX Change to strict data structures accordingly
+-- XXX Could potantially be made faster
+{-# INLINE_NORMAL isSuffixOf #-}
+isSuffixOf :: (MonadIO m, Storable a) => Stream m a -> Stream m a -> m Bool
+isSuffixOf (Stream stpa sa) (Stream stpb sb) = go1 (sa, 0, [])
```

Comment:
1) go1 is just an inefficient implementation of A.fromStream. Why not use the latter?
2) go2 is a re-implementation of lastN, why not use that API?
3) For quick check before the comparison we can use a rolling hash to reject the unsuccessful match.
4) Ideally, we do not need to buffer the entire first stream right at the start. If the second stream ends early we do not need to evaluate the rest of the first stream.